sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/json/JsonDataSource.scala

```diff
@@ -175,11 +188,15 @@ object MultiLineJsonDataSource extends JsonDataSource {
       .values
   }
 
-  private def createParser(jsonFactory: JsonFactory, record: PortableDataStream): JsonParser = {
+  private def createParser(
+      jsonFactory: JsonFactory,
+      record: PortableDataStream,
+      encoding: Option[String] = None): JsonParser = {
```

Comment:
`encoding: Option[String] = None` -> `encoding: Option[String]`. You don't have to worry about signature for private ones. Theoretically MiMa will detect the signature changes for public APIs, which make the Jenkins tests failed.